### PR TITLE
Windows event batch3 v2

### PR DESCRIPTION
**Pull request by yaelrbergman**
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
Added WindowsEvent data to the detections:
powershell_empire
GainCodeExecutionADFSviaWMI
ADFS-DKM-MasterKey-Export
AADHostLoginCorrelation
EmailAccessviaActiveSync
ExchangeServerVulnerabilitiesMarch2021IoCs
SUNSPOTLogFile

   Reason for Change(s):
Added WindowsEvent data in addition to securityEvent

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
manual testing
_Note: If updating a detection, you must update the version field._

> After the submission has been made, please look at the Validation Checks.

   Checked that the validations are passing and have addressed any issues that are present:
   -Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: Detections/MultipleDataSources/GainCodeExecutionADFSviaWMI.yaml [hunk 1]
```diff
@@ -72,6 +84,35 @@ query: |
   ) on TargetLogonId
   ),
   (
+  WindowsEvent
```
Comment (oshezaf) [inline Detections/MultipleDataSources/GainCodeExecutionADFSviaWMI.yaml:87]: Here again the original did not check for an Event Id and we may need to. I think it is also good as a best practice.

File: Detections/SecurityEvent/powershell_empire.yaml [hunk 2]
```diff
@@ -36,6 +40,29 @@ query: |
   | where EfectiveCommand matches regex regexEmpire
   | project TimeGenerated, Computer, Account = SubjectUserName, AccountDomain = SubjectDomainName, FileName = Process, EfectiveCommand, decodedCommand, encodedCommand, CommandLine, ParentProcessName
   | extend timestamp = TimeGenerated, AccountCustomEntity = Account, HostCustomEntity = Computer
+  ),
+  (WindowsEvent
+  | where EventID == 4688 and not(EventData has_any ('gc_worker.exe', 'gc_service.exe'))
+  | where EventData has_any ("-encodedCommand", "powershell.exe","powershell_ise.exe","pwsh.exe")
+  //consider filtering on filename if perf issues occur
+  //where FileName in~ ("powershell.exe","powershell_ise.exe","pwsh.exe")
```
Comment (oshezaf) [inline Detections/SecurityEvent/powershell_empire.yaml:48]: The comment would not work as is.

File: Detections/SecurityEvent/powershell_empire.yaml [hunk 3]
```diff
@@ -36,6 +40,29 @@ query: |
   | where EfectiveCommand matches regex regexEmpire
   | project TimeGenerated, Computer, Account = SubjectUserName, AccountDomain = SubjectDomainName, FileName = Process, EfectiveCommand, decodedCommand, encodedCommand, CommandLine, ParentProcessName
   | extend timestamp = TimeGenerated, AccountCustomEntity = Account, HostCustomEntity = Computer
+  ),
+  (WindowsEvent
+  | where EventID == 4688 and not(EventData has_any ('gc_worker.exe', 'gc_service.exe'))
+  | where EventData has_any ("-encodedCommand", "powershell.exe","powershell_ise.exe","pwsh.exe")
```
Comment (oshezaf) [inline Detections/SecurityEvent/powershell_empire.yaml:46]: KQL may optimze for that, but I would start with the positive tests and than do the exclusion with not.

**Review by oshezaf — CHANGES_REQUESTED** [review]
Actually it looks great. As I said: the first one is hard, but than things flow. And then, I still had comments :-)

File: Detections/MultipleDataSources/EmailAccessviaActiveSync.yaml [hunk 4]
```diff
@@ -37,6 +40,21 @@ query: |
   | project Type, TimeGenerated, Computer, Account, SubjectDomainName, SubjectUserName, Process, ParentProcessName, CommandLine
   | extend timestamp = TimeGenerated, AccountCustomEntity = Account, HostCustomEntity = Computer
   ),
+  ( WindowsEvent
```
[the remark this answers is not in the document]
Comment (yaelrbergman) [reply Detections/MultipleDataSources/EmailAccessviaActiveSync.yaml:43]: should I add event 4688 filter to security events as well? @oshezaf

[the remark this answers is not in the document]
Comment (oshezaf) [reply Detections/MultipleDataSources/EmailAccessviaActiveSync.yaml:43]: Yes

File: Detections/MultipleDataSources/ExchangeServerVulnerabilitiesMarch2021IoCs.yaml [hunk 5]
```diff
@@ -64,6 +64,13 @@ query: |
   | where ObjectName in (file_paths)
   | extend timestamp = TimeGenerated, AccountCustomEntity = Account, HostCustomEntity = Computer
   ),
+  (WindowsEvent
```
[the remark this answers is not in the document]
Comment (oshezaf) [reply Detections/MultipleDataSources/ExchangeServerVulnerabilitiesMarch2021IoCs.yaml:71]: Not sure why, but it does not show as if the version number was updated.

**Review by oshezaf — CHANGES_REQUESTED** [review]
Two comments are nice to have and one I am not sure about: it seems you missed on a version number update, though you marked it as resolved, so maybe something on my system.

**Review by oshezaf — APPROVED** [review]

**Pull request merged**